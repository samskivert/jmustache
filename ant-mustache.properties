v = V
k.v = K.V

mytrue? = true
myfalse? = false

l[01].a = L[01].A
l[01].b = L[01].B
l[02].a = L[02].A
l[02].b = L[02].B
l[03].a = L[03].A
l[03].b = L[03].B

mylist.01.prop1 = value-1-1
mylist.01.prop2 = value-1-2
mylist.02.prop1 = value-2-1
mylist.02.prop2 = value-2-2
