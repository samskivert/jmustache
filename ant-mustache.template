Hello,

1. replace value:
v = {{v}}
k.v = {{k.v}}

2. test
mytrue? = {{mytrue?}}
{{#mytrue?}}
mytrue? is valid (not false nor empty list), showing this!
{{/mytrue?}}
{{^mytrue?}}
mytrue? is NOT valid (false or empty list), showing that!
{{/mytrue?}}

myfalse? = {{myfalse?}}
{{#myfalse?}}
myfalse? is valid (not false nor empty list), showing this!
{{/myfalse?}}
{{^myfalse?}}
myfalse? is NOT valid (false or empty list), showing that!
{{/myfalse?}}

{{#doesnotexist}}
doesnotexist is valid (not false nor empty list), showing this!
{{/doesnotexist}}
{{^doesnotexist}}
doesnotexist is NOT valid (false or empty list), showing that!
{{/doesnotexist}}

3. list
{{#l}}
__id__ = {{__id__}}
a      = {{a}}
b      = {{b}}
{{/l}}
{{^l}}
l is empty or not set
{{/l}}

mylist = {{mylist}}
{{#mylist}}
{{__id__}}.prop1 = {{prop1}}
{{__id__}}.prop2 = {{prop2}}
{{/mylist}}

4. direct access:
l[02].b = {{l[02].b}}
