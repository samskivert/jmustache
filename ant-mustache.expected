Hello,

1. replace value:
v = V
k.v = K.V

2. test
mytrue? = true
mytrue? is valid (not false nor empty list), showing this!

myfalse? = false
myfalse? is NOT valid (false or empty list), showing that!

doesnotexist is NOT valid (false or empty list), showing that!

3. list
l is empty or not set

mylist = [{prop2=value-1-2, prop1=value-1-1, __id__=01}, {prop2=value-2-2, prop1=value-2-1, __id__=02}]
01.prop1 = value-1-1
01.prop2 = value-1-2
02.prop1 = value-2-1
02.prop2 = value-2-2

4. direct access:
l[02].b = L[02].B
